test/WebJobs.Extensions.Tests/Extensions/Http/ClaimsPrincipalEndToEndTests.cs

```diff
@@ -0,0 +1,206 @@
+        private const string Name = "sample@microsoft.com";
+        private const string UserIdentityJson = @"{
+  ""auth_typ"": """ + UserAuthType + @""",
+  ""claims"": [
+    {
+      ""typ"": """ + UserClaimType + @""",
+      ""val"": """ + UserClaimValue + @"""
+    },
+    {
+      ""typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+      ""val"": """ + Name + @"""
+    },
+    {
+      ""typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn"",
+      ""val"": """ + Name + @"""
+    }
+  ],
+  ""name_typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+  ""role_typ"": ""http://schemas.microsoft.com/ws/2008/06/identity/claims/role""
+}";
+
+        private const string KeyAuthType = "key";
+        private const string KeyIdClaimType = "urn:functions:keyId";
+        private const string KeyIdClaimValue = "master";
+        private const string KeyLevelClaimType = "urn:functions:authLevel";
+        private const string KeyLevelClaimValue = "admin";
+        private const string KeyIdentityJson = @"{
+  ""auth_typ"": """ + KeyAuthType + @""",
+  ""claims"": [
+    {
+      ""typ"": """ + KeyIdClaimType + @""",
+      ""val"": """ + KeyIdClaimValue + @"""
+    },
+    {
+      ""typ"": """ + KeyLevelClaimType + @""",
+      ""val"": """ + KeyLevelClaimValue + @"""
+    }
+  ],
+  ""name_typ"": ""http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"",
+  ""role_typ"": ""http://schemas.microsoft.com/ws/2008/06/identity/claims/role""
+}";
+
+        private const string OutputTestPath = @"webjobs_extensionstests\filebindinge2e_output";
+
+        private static ClaimsPrincipal globalPrincipal;
+
+        private JobHostConfiguration _config;
+        private JobHost _host;
+
+        public ClaimsPrincipalEndToEndTests()
+        {
+            var httpConfig = new HttpExtensionConfiguration();
+            _config = new JobHostConfiguration
+            {
+                TypeLocator = new ExplicitTypeLocator(typeof(TestFunctions))
+            };
+            _config.UseHttp(httpConfig);
+            _host = new JobHost(_config);
+        }
+
+        [Fact]
+        public void ClaimsPrincipalBindingWithEasyAuth()
+        {
+            var request = HttpTestHelpers.CreateHttpRequest("GET", "http://functions.com/api/TestIdentityBindings");
+            string easyAuthHeaderValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(UserIdentityJson));
+            request.Headers.Add("x-ms-client-principal", easyAuthHeaderValue);
+
+            var method = typeof(TestFunctions).GetMethod(nameof(TestFunctions.HttpRequest));
+            _host.Call(method, new { req = request });
+
+            Assert.Equal(Name, globalPrincipal.Identity.Name);
+            Assert.Equal(UserAuthType, globalPrincipal.Identity.AuthenticationType);
+            Claim testClaim = globalPrincipal.Claims.Where(claim => string.Equals(claim.Type, UserClaimType)).First();
```

Comment:
>Where [](start = 53, length = 5)

I suggest doing `FirstOrDefault(...)` instead of `Where(...).First()`. That will make the cause of a test failure more clear since it will allow you to fail at the `Assert.Equal` instead of having failing with a non-descript `InvalidOperationException`.